tf_agents/networks/multi_categorical_projection_network.py

```diff
@@ -0,0 +1,140 @@
+"""Project inputs to multiple categorical distribution object."""
+
+import gin
+import numpy as np
+import tensorflow as tf
+import tensorflow_probability as tfp
+
+from tf_agents.networks import network
+from tf_agents.networks import utils
+from tf_agents.specs import tensor_spec
+from tf_agents.specs import distribution_spec
+
+class MultiCategoricalDistributionBlock(tfp.distributions.Blockwise):
+
+  def __init__(self, logits, categories_shape):
+    self.categories_shape = categories_shape
+    distribs = self._create_distrib(logits)
+    super().__init__(distributions = distribs)
+    self._parameters['logits'] = logits
+
+  def _create_distrib(self, logits):
+    logits_splitted = tf.split(logits, self.categories_shape, -1)
+    distribs = tf.nest.map_structure(
+                    lambda l: tfp.distributions.Categorical(logits = l),
+                    logits_splitted)
+    return distribs
+
+  def _mode(self):
+    return self._flatten_and_concat_event(
```

Comment:
Why mode needs flatten_and_concat_event but sample doesn't?

Reply:
`Blockwise` from tf-probability doesn't implement `mode`, but implements `flatten_and_concat_event` for `sample` and `mean`. 

https://github.com/tensorflow/probability/blob/88d217dfe8be49050362eb14ba3076c0dc0f1ba6/tensorflow_probability/python/distributions/blockwise.py#L347-L366